common/src/main/java/gov/cms/ab2d/common/model/Contract.java

```diff
@@ -44,4 +48,35 @@
 
     @OneToMany(mappedBy = "contract")
     private Set<Coverage> coverages = new HashSet<>();
+
+    public boolean hasAttestation() {
+        return attestedOn != null;
+    }
+
+    public void clearAttestation() {
+        attestedOn = null;
+    }
+
+    public void markAttested(OffsetDateTime attestedOnDate) {
+        attestedOn = attestedOnDate;
+    }
+
+    /*
+     * Returns true if new state differs from existing which requires a save.
+     */
+    public boolean updateAttestation(boolean attested, String attestationDate, DateTimeFormatter formatter) {
+        boolean hasAttestation = hasAttestation();
+        if (attested == hasAttestation) {
+            return false;   // No changes needed
+        }
+
+        if (hasAttestation) {
+            clearAttestation();
+            return true;
+        }
+
+        LocalDateTime ldt = LocalDate.parse(attestationDate, formatter).atStartOfDay();
+        markAttested(OffsetDateTime.of(ldt, ZoneOffset.UTC)); // todo: make sure offset is correct.
```

Comment:
I think we've said that these attestations were stored in EST since CMS is in Baltimore. AttestationReportProcessor.java uses EST

Reply:
Fixed